src/Sparrow/ByteString.cs

```diff
@@ -1582,17 +1562,13 @@ private void ValidateAndUnregister(ByteStringStorage* value)
         ~ByteStringContext()
         {
             _isFinalizerThread = true;
-            try
-            {
-                Dispose();
-            }
-            catch (ObjectDisposedException)
-            {
-                // This is expected, we might be calling the finalizer on an object that
-                // was already disposed, we don't want to error here because of this
-            }
+            _wholeSegments.Clear();
```

Comment:
Why do you need this?
Can we avoid having a finalizer here entirely?

Reply:
yes, the cost of fnialization is greater then the benefit of clearing